### PR TITLE
[Agent framework] Remove langchain dependency

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/jest.config.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ module.exports = {
     '/__utils__/',
   ],
   // https://github.com/jestjs/jest/issues/6229#issuecomment-403539460
-  transformIgnorePatterns: ['node_modules/(?!langchain|langsmith)'],
+  transformIgnorePatterns: ['node_modules/(?!langsmith)'],
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply test/jest.config.js:26]: Sure, done for that.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,6 @@
     "csv-parser": "^3.0.0",
     "dompurify": "^2.4.1",
     "jsdom": "^22.1.0",
-    "langchain": "^0.0.164",
     "postinstall": "^0.7.4",
     "web-streams-polyfill": "^3.2.1"
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply package.json]: Done that.

File: test/setup.jest.ts [hunk 3]
```diff
@@ -7,7 +7,6 @@
 import { configure } from '@testing-library/react';
 import { TextDecoder, TextEncoder } from 'util';
 import 'web-streams-polyfill';
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply test/setup.jest.ts]: Done

File: server/plugin.ts [hunk 4]
```diff
@@ -14,14 +14,8 @@ import {
   Plugin,
   PluginInitializerContext,
 } from '../../../src/core/server';
-import { OpenSearchAlertingPlugin } from './adaptors/opensearch_alerting_plugin';
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply server/plugin.ts:17]: Done.

File: server/plugin.ts [hunk 5]
```diff
@@ -43,7 +37,7 @@ export class AssistantPlugin implements Plugin<AssistantPluginSetup, AssistantPl
     const openSearchObservabilityClient: ILegacyClusterClient = core.opensearch.legacy.createClient(
       'opensearch_observability',
       {
-        plugins: [PPLPlugin, OpenSearchObservabilityPlugin, OpenSearchAlertingPlugin],
+        plugins: [],
       }
     );
```
[the remark this answers is not in the document]
Comment (SuZhou-Joe) [reply server/plugin.ts]: Thanks, done for that.

File: package.json [hunk 6]
```diff
@@ -22,9 +22,7 @@
     "csv-parser": "^3.0.0",
     "dompurify": "^2.4.1",
```
Comment (wanglam) [inline package.json:23]: Shall we remove `dompurify`? It seems only be used in `server/olly/utils/output_builders/build_outputs.ts`. This file already be deleted.

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**